src/requirement-types.d.ts

```diff
@@ -7,6 +7,9 @@ type RequirementCommon = {
   readonly source: string;
   /** If this is set to true, then an edge to the course doesn't count towards double counting. */
   readonly allowCourseDoubleCounting?: true;
+  /** If this is set to true, then AP/IB credits cannot be applied towards this requirement. */
+  readonly disallowTransferCredit?: true;
```

Comment:
Not sure whether this is the right move. It increases the complexity of the requirement algorithm by introducing one more flag to worry about. While the same can be achieved by making requirements opt-in for allowing AP/IB credits by explicitly checking for AP/IB credit only course id.